src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs

```diff
@@ -223,11 +189,20 @@ public class AdminController : Controller
                 AllowHybridFlow = model.AllowHybridFlow
             };
 
-            await _applicationManager.CreateAsync(openIdApp);
+            if (model.Type == ClientType.Confidential)
+            {
+                await _applicationManager.CreateAsync(application, model.ClientSecret, HttpContext.RequestAborted);
```

Comment:
This line throws an exception if you create a new confidential app:
{System.ArgumentException: Requested value 'confidential' was not found.
   at System.Enum.TryParseEnum(Type enumType, String value, Boolean ignoreCase, EnumResult& parseResult)
   at System.Enum.Parse(Type enumType, String value, Boolean ignoreCase)
   at Orchard.OpenId.Services.OpenIdApplicationStore.SetClientTypeAsync(OpenIdApplication application, String type, CancellationToken cancellationToken) in D:\PROYECTOS\OrchardCMS\Orchard2\src\Orchard.Cms.Web\Modules\Orchard.OpenId\Services\OpenIdApplicationStore.cs:line 154
   at OpenIddict.Core.OpenIddictApplicationManager`1.<CreateAsync>d__8.MoveNext()}
Problem happens because public Task SetClientTypeAsync(OpenIdApplication application, string type, CancellationToken cancellationToken) is called with confidential type in lowercase despite Type OpenIdApplication is an enum.

I though it could be caused by this ToLower()
https://github.com/OrchardCMS/Orchard2/blob/openidUpgrade/src/Orchard.Cms.Web/Modules/Orchard.OpenId/Services/OpenIdApplicationStore.cs#L65
 but after removing it, it still happens that type received in SetClientTypeAsync is lowercase and this causes an exception
It looks openiddict provides the type in lowercase, so we have to patch it changing first letter to upper case.